ably/realtime_conn.go

```diff
@@ -312,6 +312,12 @@ func (c *Conn) isActive() bool {
 	return c.state.current == StateConnConnecting || c.state.current == StateConnConnected
 }
 
+func (c *Conn) lockCanReceiveMessages() bool {
+	c.state.Lock()
+	defer c.state.Unlock()
+	return c.state.current == StateConnConnecting || c.state.current == StateConnConnected || c.state.current == StateConnClosing
```

Comment:
This is  duplicate of `(*Conn) lockIsActive` which calls this
```
func (c *Conn) isActive() bool {
	return c.state.current == StateConnConnecting || c.state.current == StateConnConnected
}
```

So probably adding `|| c.state.current == StateConnClosing` to `(*Conn) isActive` and then calling `lockIsActive()` makes more sense in this situation.

From your pr message you can receive message when in active state so probably `lockIsActive()` alone will do?

Reply:
`isActive` determines whether you can do "active" things (ie. send messages), which you can't if you're closing. But you still can receive messages while closing, hence the different condition.

Reply:
@tcard so  you are describing `c. isActive()||c.state.current == StateConnClosing` ?

Reply:
Yes, that would be equivalent.